lambdas/src/main/kotlin/scorcerer/server/resources/League.kt

```diff
@@ -1,21 +1,69 @@
 package scorcerer.server.resources
 
+import org.http4k.core.Response
+import org.http4k.core.Status
+import org.jetbrains.exposed.sql.insert
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.transactions.transaction
 import org.openapitools.server.apis.LeagueApi
 import org.openapitools.server.models.CreateLeague200Response
 import org.openapitools.server.models.CreateLeagueRequest
 import org.openapitools.server.models.League
+import org.postgresql.util.PSQLException
+import scorcerer.server.ApiResponseError
+import scorcerer.server.db.tables.LeagueMembershipTable
+import scorcerer.server.db.tables.LeagueTable
 
 class League : LeagueApi() {
     override fun createLeague(requesterUserId: String, createLeagueRequest: CreateLeagueRequest): CreateLeague200Response {
-        TODO("Not yet implemented")
+        val id = try {
+            transaction {
+                LeagueTable.insert {
+                    it[this.name] = createLeagueRequest.leagueName
+                    it[this.id] = createLeagueRequest.leagueName.lowercase()
+                } get LeagueTable.id
+            }
+        } catch (e: PSQLException) {
+            throw ApiResponseError(Response(Status.INTERNAL_SERVER_ERROR).body("Database error"))
+        }
+        try {
+            transaction {
+                LeagueMembershipTable.insert {
+                    it[this.memberId] = requesterUserId
+                    it[this.leagueId] = id
+                }
+            }
+        } catch (e: PSQLException) {
+            if (e.message?.contains("duplicate key") == true) {
+                throw ApiResponseError(Response(Status.BAD_REQUEST).body("League already exists"))
+            } else {
```

Comment:
Don't need an else branch as we're throwing an error in the other branch

Reply:
Is this not throwing a 400 vs throwing a 500 in the else block, but as per other comment can just let 500s be caught by the `errorHandler` logic